src/containers/backpack.jsx

```diff
@@ -32,13 +44,61 @@ class Backpack extends React.Component {
             storage._hasAddedBackpackSource = true;
         }
     }
+    componentWillReceiveProps (newProps) {
+        const dragTypes = [DragConstants.COSTUME, DragConstants.SOUND];
+        if (newProps.dragInfo.dragging && !this.props.dragInfo.dragging) {
+            this.dropAreaRect = this.ref && this.ref.getBoundingClientRect();
+        } else if (!newProps.dragInfo.dragging && this.props.dragInfo.dragging && this.state.dragOver) {
```

Comment:
Comments to describe each of these cases would make this more readable

Reply:
Good call. I'll probably extract this in the next pr, when more areas become "drop areas", in a similar way to the `SortableHOC`, it'll probably become the `DropAreaHOC`. But yeah added comments, which i'll keep in if/when it is extracted.